plugin/discovery/find.go

```diff
@@ -96,6 +96,7 @@ func findPluginPaths(kind string, machineName string, dirs []string) []string {
 				continue
 			}
 
+			// FIXME: we pass in GOOS_GOARCH paths directly, so these may not be "legacy"
```

Comment:
What I mean is that we are currently passing in both directories that _contain_ an `os_arch` subdir which we descend into here, as well as the `os_arch` directory itself which picks up the versioned plugins in this "legacy"  block below. I made this note to remember to look for and fix the call-sites so that we they deal directly with the `os_arch` directories, and let them be discovered by this function.